applications/HDF5Application/python_scripts/create_xdmf_file.py

```diff
@@ -32,17 +32,20 @@ def GetSpatialGrid(h5py_file):
 
 
 def GetNodalResults(h5py_file):
-    nodal_results_path = "/ResultsData/NodalResults"
+    nodal_results_paths = ["/ResultsData/NodalSolutionStepData", "/ResultsData/NodalDataValues"]
     results = []
```

Comment:
You mentioned there is a problem if the same variable is used in data values and nodal solution step values. To detect this, can we replace the `results = []` with `results = {}` and replace this function body with

```
results = {}
if "/ResultsData/NodalSolutionStepData" in h5py_file:
    AddNodalSolutionStepData(h5py_file.get("..."), results)
if "/ResultsData/NodalDataValues" in h5py_file:
    AddNodalDataValues(h5py_file.get("..."), results)
return list(results.values)
```

and in the `Add...` functions do 

```
if variable_name in results:
    raise ValueError('Nodal result "' + variable_name + '" is already defined.')
```